coral/src/app/features/topics/request/TopicRequest.test.tsx

```diff
@@ -262,78 +487,38 @@ describe("<TopicRequest />", () => {
       );
-      // Wait environments to be loaded
-      await screen.findByLabelText("Environment");
-    });
-    afterAll(() => {
-      cleanup();
     });
+    afterAll(cleanup);
 
     describe("input components", () => {
-      it('should render <select /> when "maxReplicationFactor" is defined', async () => {
-        await user.selectOptions(screen.getByLabelText("Environment"), "PROD");
-        screen.getByRole("combobox", { name: "Replication factor" });
-      });
-
-      it('should render <input type="number" /> when "maxReplicationFactor" not defined', async () => {
-        await user.selectOptions(screen.getByLabelText("Environment"), "DEV");
-        screen.getByRole("spinbutton", { name: "Replication factor" });
-      });
-    });
-
-    describe("when environment is changed", () => {
-      it('changes replication factor value to environment "defaultPartitions"', async () => {
-        await user.selectOptions(
-          screen.getByLabelText("Environment"),
-          "WITH_DEFAULT_PARTITIONS"
-        );
-        await waitFor(() => {
-          expect(
-            screen.getByLabelText("Replication factor")
-          ).toHaveDisplayValue("4");
+      it('should render <select /> when "maxPartitions" is defined', async () => {
+        const selectEnvironment = await screen.findByRole("combobox", {
+          name: "Environment",
         });
-      });
-
-      it('changes replication factor value to environment "maxPartitions" when exceeded it is and no default', async () => {
-        await user.selectOptions(screen.getByLabelText("Environment"), "DEV");
-        await user.clear(screen.getByLabelText("Replication factor"));
-        await user.type(screen.getByLabelText("Replication factor"), "100");
-        expect(screen.getByLabelText("Replication factor")).toHaveDisplayValue(
-          "100"
-        );
+        await user.selectOptions(selectEnvironment, "PROD");
 
-        await user.selectOptions(screen.getByLabelText("Environment"), "TST");
-        await waitFor(() => {
-          expect(
-            screen.getByLabelText("Replication factor")
-          ).toHaveDisplayValue("2");
+        const topicPartitionSelect = screen.getByRole("combobox", {
+          name: "Topic partitions",
         });
+        expect(topicPartitionSelect).toBeEnabled();
       });
 
-      it('keeps topic replication factor value if not default and value does not exceeded "maxPartitions"', async () => {
-        await user.selectOptions(screen.getByLabelText("Environment"), "PROD");
-        await user.selectOptions(
-          screen.getByLabelText("Replication factor"),
-          "4"
-        );
-        await waitFor(() => {
-          expect(
-            screen.getByLabelText("Replication factor")
-          ).toHaveDisplayValue("4");
+      it('should render <input type="number" /> when "maxPartitions" is not defined', async () => {
+        const selectEnvironment = await screen.getByRole("combobox", {
```

Comment:
Is there a reason for this one to `await` `getByRole` instead of `findByRole` as we see here? https://github.com/aiven/klaw/pull/446/files#diff-9c73423f6e6369621f0951894ae4f5fd4ee717f8d5740cbe896ca6683d3f5c70R576

And actually, does it do anything? `getByRole` does not return a promise :o
![Screenshot 2023-01-19 at 09 02 05](https://user-images.githubusercontent.com/20607294/213387128-406e666f-b27b-4dfb-a05d-d6ee431e95fa.png)

Not a passive aggressive comment, naive curiosity ^^

Reply:
No, that's an error, it should be `findBy` 🙈  
There's already a housekeeping ticket to add more linters and that should prevent this in the future.